weather_dl/download_pipeline/clients.py

```diff
@@ -209,6 +215,105 @@ def num_requests_per_key(self, dataset: str) -> int:
         return 2
 
 
+class EumetsatClient(Client):
+    def __init__(self, config: Config, level: int = logging.INFO,
+                 config_subsection: t.Optional[t.Tuple] = None) -> None:
+        super().__init__(config, level)
+        self.key = config.kwargs.get('api_key', os.environ.get("EUMETSATAPI_KEY"))
+        self.secret = config.kwargs.get('api_secret', os.environ.get("EUMETSATAPI_SECRET"))
+        if not self.key and config_subsection:
+            self.key = config_subsection[1].get('api_key')
+        if not self.secret and config_subsection:
+            self.secret = config_subsection[1].get('api_secret')
```

Comment:
Why do we need to pass in a config subsection here? Shouldn't this already be done in the partition step?  The standard key is the one from the subsection.